tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -289,6 +300,12 @@ private async Task RestorePinnedWidgetsAsync(ComSafeWidget[] hostWidgets)
         // append it at the end. If a position is missing, just show the next widget in order.
         foreach (var widget in hostWidgets)
         {
+            if (cancellationToken.IsCancellationRequested)

```

Comment:
Curious what's the crash about? Assuming some resources are disposed? Nonetheless, this likely greatly reduces the crash but may not fully eliminate it depending on the concurrency execution, for example if `IsCancellationRequested` was set right after the `if` statement. I see that you've linked some related tasks to further strengthen this 👍

Reply:
It's `System.InvalidOperationException: Collection was modified; enumeration operation may not execute.` Specifically, in OnUnloaded we do PinnedWidgets.Clear() and it was modifying PinnedWidgets. So if OnLoaded was still running, the iterators over PinnedWidgets inside RestorePinnedWidgetsAsync() were now invalid and throwing.
It's specifically the case where we leave Dashboard while it's still loading.

Reply:
Yeah, I considered just locking but I'd really prefer to bail early if possible since it's really a lot of work that gets us nothing. I looked into locking as well, but since it should go everywhere I think that deserves its own PR (for the issue linked)